index.js

```diff
@@ -864,6 +864,45 @@
 
   //. ### List
 
+  //# slice :: Number -> Number -> [a] -> Maybe [a]
+  //.
+  //. Returns Just a list containing the elements from the supplied list
+  //. from a beginning index (inclusive) to an end index (exclusive).
+  //. Returns Nothing unless the start interval is less than or equal to
+  //. the end interval, and the list contains both (half-open) intervals.
+  //. Accepts negative indices, which indicate an offset from the end of
+  //. the list.
+  //.
+  //. ```javascript
+  //. > S.slice(1, 1, ['a', 'b', 'c', 'd', 'e'])
+  //. Just([])
+  //.
+  //. > S.slice(1, 3, ['a', 'b', 'c', 'd', 'e'])
+  //. Just(["b", "c"])
+  //.
+  //. > S.slice(-2, -0, ['a', 'b', 'c', 'd', 'e'])
+  //. Just(["d", "e"])
+  //.
+  //. > S.slice(2, -0, ['a', 'b', 'c', 'd', 'e'])
+  //. Just(["c", "d", "e"])
+  //.
+  //. > S.slice(0, 2, [])
+  //. Nothing()
+  //.
+  //. > S.slice(1, 6, ['a', 'b', 'c', 'd', 'e'])
+  //. Nothing()
+  //. ```
+  var slice = S.slice = R.curry(function(start, end, xs) {
+    var len = xs.length;
+    var startIdx = R.eq(-0, start) ? len : start < 0 ? start + len : start;
+    var endIdx = R.eq(-0, end) ? len : end < 0 ? end + len : end;
+
+    if (Math.abs(start) > len || Math.abs(end) > len || startIdx > endIdx) {
+      return Nothing();
+    }
+    return Just(R.slice(startIdx, endIdx, xs));
```

Comment:
I like to avoid using `return` for control flow where possible. Let's write something like this instead:

```javascript
return (Math.abs(start) <= len &&
        Math.abs(end) <= len &&
        startIdx <= endIdx) ?
  Just(R.slice(startIdx, endIdx, xs)) :
  Nothing();
```

I'm sorry about the nitpicking!